src/components/atoms/IconInput/IconInput.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { ChangeEvent } from "react";
+import classNames from "classnames";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import "./IconInput.scss";
+
+export interface IconInputProps {
+  className?: string;
+  icon?: IconProp;
+  onChange?: (e?: ChangeEvent<HTMLInputElement>) => void;
+  value?: string;
+  autoComplete?: string;
+  placeholder?: string;
+  iconPosition?: "right" | "left";
+}
+
+const ICON_INPUT_CLASS_NAME = "icon-input";
+const ICON_CLASS_NAME = `${ICON_INPUT_CLASS_NAME}__icon`;
+const INPUT_CLASS_NAME = `${ICON_INPUT_CLASS_NAME}__input`;
+
+export const IconInput: React.FC<IconInputProps> = ({
+  className,
+  icon,
+  onChange,
+  value,
+  autoComplete = "off",
+  placeholder,
+  iconPosition = "left",
+}) => {
+  const wrapperClassNames = classNames(ICON_INPUT_CLASS_NAME, className);
+  const iconClassNames = classNames(
+    ICON_CLASS_NAME,
+    iconPosition === "right" && `${ICON_CLASS_NAME}--right`
+  );
+  const inputClassNames = classNames(
+    INPUT_CLASS_NAME,
+    `${INPUT_CLASS_NAME}--icon-from-${iconPosition}`
+  );
+
+  return (
+    <div className={wrapperClassNames}>
+      <input
+        className={inputClassNames}
+        placeholder={placeholder}
+        value={value}
+        onChange={onChange}
+        autoComplete={autoComplete}
+      />
```

Comment:
We may want to be able to pass `ref` to this input so that we can use it with `react-hook-form` / `useForm`s `register` function and similar.

This might be useful reading for this:

- https://reactjs.org/docs/forwarding-refs.html

Reply:
It might also be nice to be able to pass any `extraProps` through to this without them necessarily being explicitly defined by name in `IconInputProps`. We can use the `{...extraInputProps}` syntax to add that here (and would also need a `...extraInputProps` as the last value in the function props definition) Not sure what the exact typing will be to represent that in `IconInputProps`, but it should be relatively easy to lookup (or if there is a specific type for 'all the things that can be on an input field', that would be a useful type to use there maybe)

That'll save us having to explicitly add every single prop we would ever want to be able to pass through to this component individually.

Reply:
forwarding Ref: https://github.com/sparkletown/sparkle/pull/1306/commits/c137aec001cc7119da4274f039102d1d65e2c96b
applying extra props: https://github.com/sparkletown/sparkle/pull/1306/commits/4522f27ae790a9bc30e71c2e4e06433f7e2b9141
and renaming inputProps -> extraInputProps: https://github.com/sparkletown/sparkle/pull/1306/commits/2d98fa288fa22220801a9a7f3765e88ec04a549c